File: gitea_repos_master/isphere-classes/worker-classes/worker_classes/keydb/response_builder.py
import datetime
import logging

from worker_classes.keydb.interfaces import KeyDBResponse, Records


class KeyDBResponseBuilder:
    @classmethod
    def __base_response(
        cls, code=200, message="ok", status="ok", records=None
    ) -> KeyDBResponse:
        if records is None:
            records = []
        logging.info(f"KeyDB response: [{code}] {message}")
        return {
            "status": status,
            "code": code,
            "message": str(message),
            "records": records,
            "timestamp": int(datetime.datetime.now().timestamp()),
        }

    @staticmethod
    def ok(response: Records) -> KeyDBResponse:
        return KeyDBResponseBuilder.__base_response(200, records=response)

    @staticmethod
    def empty() -> KeyDBResponse:
        return KeyDBResponseBuilder.__base_response(204)

    @staticmethod
    def no_sessions() -> KeyDBResponse:
        return KeyDBResponseBuilder.__base_response(
            512, status="Error", message="Нет активных сессий"
        )

    @staticmethod
    def error(e: Exception, code=500) -> KeyDBResponse:
        return KeyDBResponseBuilder.__base_response(code, status="Error", message=str(e))

File: gitea_repos_master/whatsapp/whatsapp-baileys-api/src/logic/keydb/response_builder.ts
import { Records } from 'keydb';
import logger from '../logger/winston';

export class KeyDBResponseBuilder {
  private static __base_response(
    code: number = 200,
    message: string = 'ok',
    status: string = 'ok',
    records: Records | null = null
  ) {
    if (!records) {
      records = [];
    }

    logger.info(`KeyDB response: [${code}] ${String(message)}`);
    return {
      status,
      code,
      timestamp: Math.floor(Date.now() / 1000),
      message: String(message),
      records
    };
  }

  public static ok(response: Records) {
    return KeyDBResponseBuilder.__base_response(200, 'ok', 'ok', response);
  }

  public static empty() {
    return KeyDBResponseBuilder.__base_response(204);
  }

  public static error(e: Error | string, code: number = 500) {
    return KeyDBResponseBuilder.__base_response(code, String(e), 'Error');
  }
}
